<?xml version="1.0" ?>
<robot name="quadro" xmlns:xacro="http://www.ros.org/wiki/xacro" >

    <ros2_control name="CybergearActuators" type="actuator">
    <hardware>
      <plugin>quadro/CybergearActuator</plugin>
      <param name="can_interface">can0</param>
      <param name="timeout_sec">3.0</param>
      <param name="use_bus_time">false</param>
      <param name="interval_sec">0.5</param>
      <param name="primary_id">0</param>
    </hardware>

    <joint name="br_m1_s1">
      <command_interface name="position">
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="br_m2_s2">
      <command_interface name="position">
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="br_m3_s3">
      <command_interface name="position">
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>

    <joint name="bl_m1_s1">
      <command_interface name="position">
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="bl_m2_s2">
      <command_interface name="position">
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="bl_m3_s3">
      <command_interface name="position">
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>

    <joint name="fr_m1_s1">
      <command_interface name="position">
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="fr_m2_s2">
      <command_interface name="position">
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="fr_m3_s3">
      <command_interface name="position">
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
  

    <joint name="fl_m1_s1">
      <command_interface name="position">
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="fl_m2_s2">
      <command_interface name="position">
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="fl_m3_s3">
      <command_interface name="position">
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
  </ros2_control>

</robot>